### PR TITLE
Added some error messages and tracing to bridge network initialization

### PR DESCRIPTION
**Pull request by johngossman**
Signed-off-by: John Gossman johngos@microsoft.com

### CODE REVIEW

**Comment by johngossman** [comment]
I'm trying to debug docker/docker#12177. No luck so far, but this level of tracing will be helpful to find repros.

**Comment by johngossman** [comment]
I'm not married to any of the wording or where I put messages. Looking for feedback.

File: daemon/daemon.go [hunk 1]
```diff
@@ -938,6 +938,7 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine, registryService
 
 	if !config.DisableNetwork {
 		if err := bridge.InitDriver(&config.Bridge); err != nil {
+			logrus.Errorf("Error initializing Bridge: %s", err)
```
Comment (ahmetalpbalkan) [inline daemon/daemon.go]: I think this should be `return nil, fmt.Errorf("Error initializing Bridge: %v", err)` (note the %v).

The handler at the top level can then show where exactly this is coming from.

File: daemon/networkdriver/bridge/driver.go [hunk 2]
```diff
@@ -135,8 +135,11 @@ func InitDriver(config *Config) error {
 			return err
 		}
 
+		logrus.Infof("Bridge interface not found, trying to create it")
```
Comment (ahmetalpbalkan) [inline daemon/networkdriver/bridge/driver.go]: sounds more like .Debug (also shouldn't end with 'f' like Infof)

File: daemon/networkdriver/bridge/driver.go [hunk 3]
```diff
@@ -272,6 +277,7 @@ func InitDriver(config *Config) error {
 		}
 		logrus.Debugf("Subnet: %v", subnet)
 		if err := ipAllocator.RegisterSubnet(subnet, subnet); err != nil {
+			logrus.Errorf("Error registering subnet for IPv4 bridge network: %s", err)
```
Comment (ahmetalpbalkan) [inline daemon/networkdriver/bridge/driver.go]: v6?

File: daemon/daemon.go (hunk 1 above)
Comment (johngossman) [inline daemon/daemon.go]: Fixed

Comment (ahmetalpbalkan) [inline daemon/daemon.go]: @johngossman can we fix other log-then-return(s) as well?

**Comment by ahmetalpbalkan** [comment]
It'd be better to add new detail messages to the returned errors instead if logging to output streams.

**Comment by johngossman** [comment]
@ahmetalpbalkan I guess the question is philosophical: is the error returned an aggregate of all the calls that failed, or something easy to read for a user. If the latter, then logging makes more sense. Docker codebase and even golang don't seem to have standard best practices....or at least I don't know them.

**Comment by crosbymichael** [comment]
LGTM

**Comment by ehazlett** [comment]
LGTM

**Comment by ahmetalpbalkan** [comment]
@ehazlett note for the future, we should ask contributors to squash commits into one where possible http://docs.docker.com/project/create-pr/

**Comment by ehazlett** [comment]
@ahmetalpbalkan thanks for the heads up!

**Comment by johngossman** [comment]
@ehazlett Thanks for the merge!

@ahmetalpbalkan ps -- I would have squashed, was just waiting to see if I got more feedback.

**Pull request merged**